util/util.go

```diff
@@ -0,0 +1,82 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	//[2016-05-06T20:04:23+00:00]
+	longForm = "[2006-01-02T15:04:05-07:00]"
+)
+
+func fromKubernetes(json map[string]interface{}) bool {
+	return json["kubernetes"] != nil
+}
+
+// FromContainer checks that the log message is from a given container
+func FromContainer(json map[string]interface{}, pattern string) bool {
+	if !fromKubernetes(json) {
+		return false
+	}
+
+	containerName := json["kubernetes"].(map[string]interface{})["container_name"].(string)
+	matched, _ := regexp.MatchString(pattern, containerName)
+	return matched
+}
+
+//ParseMessage takes a string and returns a map
+func ParseMessage(message string) (map[string]interface{}, error) {
+	curlyIndex := strings.Index(message, "{")
+	if curlyIndex > -1 {
+		message = message[curlyIndex:]
+		var messageJSON map[string]interface{}
+		err := json.Unmarshal([]byte(message), &messageJSON)
+		if err == nil {
+			return messageJSON, nil
+		}
+		return nil, err
+	}
+	return nil, fmt.Errorf("Not a valid JSON message: %s", message)
+}
+
+//GetHost returns the host value from the kubernetes submap
+func GetHost(json map[string]interface{}) string {
+	return json["kubernetes"].(map[string]interface{})["host"].(string)
+}
+
+//ParseNginxLog returns map of the parsed nginx log
+func ParseNginxLog(message string) (map[string]interface{}, error) {
+	parsedMessage := make(map[string]interface{})
+	splitMessage := strings.Split(message, " - ")
+	if len(splitMessage) > 1 {
+		parsedMessage["time"] = toTime(splitMessage[0])
+		parsedMessage["app"] = strings.TrimSpace(splitMessage[1])
+		parsedMessage["status_code"] = strings.TrimSpace(splitMessage[3])
+		parsedMessage["bytes_sent"] = parseInt(strings.TrimSpace(splitMessage[5]))
+		parsedMessage["response_time"] = parseFloat(strings.TrimSpace(splitMessage[11]))
+		parsedMessage["request_time"] = parseFloat(strings.TrimSpace(splitMessage[12]))
+		return parsedMessage, nil
+	}
+	return nil, fmt.Errorf("Invalid nginx log message: %s", message)
+}
+
+//Turn this [2016-05-06T20:04:23+00:00] into time object
+func toTime(date string) time.Time {
+	t, _ := time.Parse(longForm, date)
+	return t
+}
+
+func parseInt(s string) int {
+	v, _ := strconv.Atoi(s)
+	return v
+}
+
+func parseFloat(s string) float64 {
+	v, _ := strconv.ParseFloat(s, 64)
+	return v
+}
```

Comment:
particularly because i loathe double returns liek this because i cant one line an assignment..